src/main/kotlin/dev/arbjerg/lavalink/client/LavalinkNodeOptions.kt

```diff
@@ -0,0 +1,39 @@
+package dev.arbjerg.lavalink.client
+
+import dev.arbjerg.lavalink.client.loadbalancing.IRegionFilter
+import dev.arbjerg.lavalink.internal.TIMEOUT_MS
+import java.net.URI
+
+data class LavalinkNodeOptions(val name: String,
+                               val serverUri: URI,
+                               val password: String,
+                               val regionFilter: IRegionFilter?,
+                               val httpTimeout: Long) {
+    data class Builder(
+        var name: String? = null,
+        private var serverUri: URI? = null,
+        private var password: String? = null,
+        private var regionFilter: IRegionFilter? = null,
+        private var httpTimeout: Long = TIMEOUT_MS,
+    ) {
+        fun name(name: String) = apply { this.name = name }
+        fun serverUri(serverUriString: String) = apply { this.serverUri = URI(serverUriString) }
+        fun serverUri(serverUri: URI) = apply { this.serverUri = serverUri }
+        fun password(password: String) = apply { this.password = password }
+        fun regionFilter(regionFilter: IRegionFilter?) = apply { this.regionFilter = regionFilter }
+        fun httpTimeout(httpTimeout: Long) = apply { this.httpTimeout = httpTimeout }
```

Comment:
If you make all the vars public, you don't need these. Also please stick to java naming conventions for setter methods

Reply:
Is it typical for builders to do that? I dont think I have seen a builder that has you do direct access like that, given I dont use kotlin all that much.

Reply:
Yeah ignore that. if you make the vars public you can't do chaining. 
What I do want is for you to rename the setters to `setX`